beacon_node/network/src/router/gossip_processor.rs

```diff
@@ -0,0 +1,810 @@
+                {
+                    error!(
+                        self.log,
+                        "Aggregate attestation queue full";
+                        "msg" => "the system has insufficient resources for load",
+                        "queue_len" => aggregate_queue.max_length,
+                    )
+                }
+
+                if initial_attestation_queue_len != attestation_queue.len()
+                    && attestation_queue.is_full()
+                {
+                    error!(
+                        self.log,
+                        "Attestation queue full";
+                        "msg" => "the system has insufficient resources for load",
+                        "queue_len" => attestation_queue.max_length,
+                    )
+                }
+            }
+        };
+
+        // Spawn on the non-blocking executor.
+        executor.spawn(manager_future, MANAGER_TASK_NAME);
+
+        event_tx
+    }
+
+    /// Spawns a blocking worker thread to process some `Work`.
+    ///
+    /// Sends an message on `idle_tx` when the work is complete and the task is stopping.
+    fn spawn_worker(
+        &mut self,
+        mut idle_tx: mpsc::Sender<()>,
+        message_id: MessageId,
+        peer_id: PeerId,
+        work: Work<T::EthSpec>,
+    ) {
+        let worker_timer = metrics::start_timer(&metrics::GOSSIP_PROCESSOR_WORKER_TIME);
+        metrics::inc_counter(&metrics::GOSSIP_PROCESSOR_WORKERS_SPAWNED_TOTAL);
+
+        self.current_workers = self.current_workers.saturating_add(1);
+        let chain = self.beacon_chain.clone();
+        let network_tx = self.network_tx.clone();
+        let sync_tx = self.sync_tx.clone();
+        let log = self.log.clone();
+        let executor = self.executor.clone();
+
+        executor.spawn_blocking(
+            move || {
+                let _worker_timer = worker_timer;
+
+                // We use this closure pattern to avoid using a `return` that prevents the
+                // `idle_tx` message from sending.
+                let handler = || {
+                    match work {
+                        /*
+                         * Unaggregated attestation verification.
+                         */
+                        Work::Attestation(boxed_tuple) => {
+                            let (attestation, subnet_id, should_import) = *boxed_tuple;
+
+                            let _attestation_timer = metrics::start_timer(
+                                &metrics::GOSSIP_PROCESSOR_UNAGGREGATED_ATTESTATION_WORKER_TIME,
+                            );
+                            metrics::inc_counter(
+                                &metrics::GOSSIP_PROCESSOR_UNAGGREGATED_ATTESTATION_VERIFIED_TOTAL,
+                            );
+
+                            let beacon_block_root = attestation.data.beacon_block_root;
+
+                            let attestation = if let Ok(attestation) = chain
```

Comment:
This `if let Ok(_)` makes it look like we're dropping the error message, might be clearer to write it as a match that plumbs the error into `handle_attestation_verification_failure`?

Reply:
I won't block on it though